### ISSUE TITLE
Problems using winrt::Windows::Storage::Streams::DataWriter.StoreAsync()

### ISSUE THREAD

**Issue by daijava666** [cgsaxner/HoloLens2-Unity-ResearchModeStreamer#16]
Hello @cgsaxner ，
I’m sorry to ask you about this basic question, but I’m really desperate.
After reading your code, I want to send the IMU data to the PC via TCP/IP, so I wrote a data sending function, and the code is as follows. 
If I add m_writer.StoreAsync(); this sentence to the code, the whole program will crash during the debugging process. If it is not added, it can be executed. The problem is as follows.

I wrote according to your method, or according to the official method of Microsoft, it is of no avail, so I come to ask about the usage of this sentence[m_writer.StoreAsync();], I hope to get your reply, thanks for a lot.

(1)code

```
void AccelProcessor::Send(DirectX::XMFLOAT3 m_accelSample)
{
    wchar_t msgBuffer[200];

    OutputDebugString(L"AccelProcessor::Send: Start Send Data\n");

    float accelx = m_accelSample.x;
    float accely = m_accelSample.y;
    float accelz = m_accelSample.z;

    swprintf_s(msgBuffer, L"####Accel: % 3.4f % 3.4f % 3.4f\n", accelx, accely, accelz);

    OutputDebugStringW(msgBuffer);

    try
    {
        m_writer.WriteDouble(accelx);
        m_writer.WriteDouble(accely);
        m_writer.WriteDouble(accelz);

        m_writer.StoreAsync();
    }

    catch (winrt::hresult_error const& ex)
    {
        SocketErrorStatus webErrorStatus{ SocketError::GetStatus(ex.to_abi()) };
        if (webErrorStatus == SocketErrorStatus::ConnectionResetByPeer)
        {
            // the client disconnected!
            m_writer == nullptr;
            m_streamSocket == nullptr;
        }
#if DBG_ENABLE_ERROR_LOGGING
        winrt::hstring message = ex.message();
        OutputDebugStringW(L"AccelProcessor::Send: Sending failed with ");
        OutputDebugStringW(message.c_str());
        OutputDebugStringW(L"\n");
#endif // DBG_ENABLE_ERROR_LOGGING
    }

}
```

(2)problem
```
1.WinRT originate error - 0x800710DD : 'The operation identifier is not valid.';
2.Exception thrown at 0x00007FF9261E48CC (located in imu.exe): Microsoft C++ exception: winrt::hresult_error, located in memory location;
3.Exception raised at 0007FF9261E48CC (located in imu.exe): Microsoft C++ exception: winrt::hresult_error, located at memory location 0x0000007B39BFF5E0;
```

**Comment by daijava666** [comment]
The function StartServer() and OnConnectionReceived() are same as your code.

**Comment by cgsaxner** [comment]
Hi @daijava666,
have you tried writing the values to the output stream using `m_writer.WriteSingle()`? Your acceleration values are single floating point, yet you write them as double floating point values, which might cause problems.

**Comment by daijava666** [comment]
Hi @cgsaxner ，
I have solded the above problem, but now I have another problem, I would like to ask you about it. I integrated your code with the MRTK tool and ran it on Hololens2. It is found that as the program runs, as long as the image is transferred, the MRTK tool will prompt that the available memory is rapidly reduced, and the entire program will crash due to memory overflow. I am wondering if it is because the transmission speed of TCP/IP is not as fast as the sampling speed of the sensor, which results in the accumulation of data and the reduction of memory. Have you ever done this?
